sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/ProtobufUtil.java

```diff
@@ -44,6 +39,43 @@ private ProtobufUtil() {
     }
   }
 
+  public static <MESSAGE extends Message> Iterable<MESSAGE> readFileMessages(File file, final Parser<MESSAGE> parser) {
+    try {
+      final InputStream inputStream = new BufferedInputStream(new FileInputStream(file));
+      return new Iterable<MESSAGE>() {
+        @Override
+        public Iterator<MESSAGE> iterator() {
```

Comment:
It is better to construct the inputStream here so that Iterable is "reentrant".
Iterable iterable = xxx;
iterable.iterator().next() => return first element
iterable.iterator().next() => return first element